src/res_scanner.ml

```diff
@@ -487,11 +487,11 @@ let scanTemplateLiteralToken scanner =
       | _ ->
         next scanner;
         scan ())
-    | ch when ch == hackyEOFChar ->
+    | ch when ch = hackyEOFChar ->
       let endPos = position scanner in
       scanner.err ~startPos ~endPos Diagnostics.unclosedTemplate;
       Token.TemplateTail(
-        (String.sub [@doesNotRaise]) scanner.src startOff (scanner.offset - 1 - startOff)
+        (String.sub [@doesNotRaise]) scanner.src startOff (max (scanner.offset - 1 - startOff) 0)
```

Comment:
for cases like
```
`${`
```
for the inner `, it is already the end of file, here both `startOff` and `offset` points to the end of the file, so the length was actually `-1`, let me know if you have a better fix